identity-flow/src/main/java/com/android/identity/flow/handler/FlowNotificationsLocal.kt

```diff
@@ -35,10 +35,17 @@ class FlowNotificationsLocal(
         // TODO: it is not quite clear if we should delay notifications. It might make
         // sense to actually remove this and ensure that everything works as expected,
         // as the need for the delay is an indication of a race condition somewhere
+        // Update: it seems that we start listening for notifications a bit later than
```

Comment:
I think we can adjust it, yes. This is very much related to notification reliability (which was one of your other comments). Since notifications are best effort we *probably* should also poll a bit in certain situations instead of relying purely on notifications. That code is in DocumentModel and I'd rather have a more surgical PR to tweak it. Alternatively, we could have a more reliable notification contract and keep them in the queue for some time, that way it would be OK to do an operation that causes notification and then immediately start listening on it. I am kinda leaning that way, actually.

Reply:
k, sg. This is fine for now. We can refactor in the future.